running-code/execute-package.md

```diff
@@ -0,0 +1,163 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.16.4
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+(execute-package)=
+# How to Execute a Python Package
+
+In [How to Execute a Python Script](execute-script) you learned about two primary ways to execute a stand-alone Python script.
+There are two other ways to execute Python code as commands, both of which work for code that has been formatted as a package.
+
+1. You can execute modules using their import name
+2. A package can provide arbitraty command names that execute parts of themselves
+
+## 1. Executable modules
+
+In the [Pass your script to the Python command lesson](execute-script-pass-to-python) you learned that the `python` command can
+be passed a file path for execution. Alternatively you can also pass the name of a module, exactly as would be used after an `import`.
+In this case, Python will look up the module referenced in the
+[currently active environment](https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/#create-and-use-virtual-environments),
+and will execute it as a script.
+
+This execution mode is performed with the `-m` flag, as in `python -m site`. It can be used in place of a file
+path, but cannot be used in combination with a path, as there can only be one executing module.
+
+:::{tip}
+These commands both do the same thing, but the latter is easier to remember and much more portable, as it doesn't rely
+on the local machine's file system details.
+
+```bash
+python ./.venv/lib/python3.12/site-packages/pip/__main__.py
+```
+
+```bash
+python -m pip
+```
+:::
+
+### Further exploration
+
+On your own or in small groups:
+
+* Install the `my_program.py` module from the [last lesson](execute-script-launch-command)
+* Try to get the same greeting as before using `python -m my_program`.
+
+**my_program.py** which prints out `✨ Hello from Python ✨` when executed as a script.
+
+```python
+#!/usr/bin/env python
+# The above line is a shebang and can take the place of typing python on the command line
+# This comment is below, because shebangs must be the first line!
```

Comment:
```suggestion
# Remember our use of shebangs from the executing python scripts lesson?
```

something like calling back to where this was described, since there is more detail there, and it feels odd to repeat it without reference to the prior explanation

Reply:
this suggestion is bad but yno what i mean

Reply:
took it out as I think it is just distracting. I don't care at this point in the lesson if the reader know what a shebang is or why its there. I just want an example that will work as demonstrated (calling the file as a command).